[Timothy Putra Prasetio] iP

tim-pipi opened the pull request:
# Anya Forger - Your Task Management Chatbot

Anya Forger is a chatbot inspired by the character from the [_Spy x Family_](https://en.wikipedia.org/wiki/Spy_%C3%97_Family) anime. She is your ultimate task management companion. Whether you're a secret agent or just someone with a busy schedule, Anya can help you stay organized and on top of your tasks.

## Features:

* **Task Management**: Anya Forger can create, edit, and track your tasks effortlessly.
* **Priority Sorting**: Arrange your tasks by priority to ensure you focus on what matters most.
* **Deadline Alerts**: Receive timely reminders so you never miss a deadline.
* **Cross-Device Sync**: Access your tasks from anywhere, on any device.

## 🚀Mission Accomplished:
> "Anya Forger is the secret agent of Task Management." - Anonymous 

## Task List:
- [ ] Finish coding the chatbot
- [ ] Test Anya's task management features
- [ ] Write documentation
- [ ] Deploy Anya to production!

## Getting Started:

1. Download the `anya.jar` file from [here]()
1. Run the following command:
```
java -jar Anya.jar 
```
> ✏️ Note: run the command in the same folder as the jar file

If you are Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

tim-pipi committed:
Update .gitignore to ignore .class files

tim-pipi committed:
Renamed Duke to Anya Forger and added Greeting and Exit

tim-pipi committed:
Update .gitignore

tim-pipi committed:
Add Echo function

tim-pipi committed:
Add list command

tim-pipi committed:
Add Task class

tim-pipi committed:
Add Mark as Done function

tim-pipi committed:
Add three new types of Task: Todo, Deadline, Event

tim-pipi committed:
Support multiple types of tasks

tim-pipi committed:
Add Automated Text UI Testing files

tim-pipi committed:
Refactor name: Duke to Anya

tim-pipi committed:
Add Error Handling and Refactor Code: Encapsulate common patterns in functions

tim-pipi committed:
Add Delete Function

tim-pipi committed:
Add Enums

tim-pipi committed:
Add Save Functionality

tim-pipi committed:
Merge from branch-level-7

tim-pipi committed:
Change date and time to use LocalDateTime

tim-pipi committed:
Merge branch 'branch-level-8'

tim-pipi committed:
Add MoreOOP files and directories

tim-pipi committed:
Add Storage class methods

tim-pipi committed:
Refactor Storage.java

tim-pipi committed:
Add TaskList

tim-pipi committed:
Add More OOP

tim-pipi committed:
Merge branch 'branch-A-MoreOOP'

tim-pipi committed:
Merge branch 'add-gradle-support'

tim-pipi committed:
Setup gradle to automate build

tim-pipi committed:
"Add checkstyle and JUnit test"

tim-pipi committed:
Add JAR

tim-pipi committed:
Add JavaDoc header comments

tim-pipi committed:
Follow Coding Standard

tim-pipi committed:
Add new Command FIND

tim-pipi committed:
Add find functionality

tim-pipi committed:
Merge branch 'branch-A-JavaDoc'

tim-pipi committed:
Fix merge conflicts

tim-pipi committed:
Fix merge conflicts

FerdiHS commented on data/tasks.txt:
Shouldn't this file be ignored?

FerdiHS commented on src/main/java/anya/Anya.java:
```diff
@@ -0,0 +1,68 @@
+     * @see Storage
+     * @see Parser
+     * @see InvalidStorageFilePathException
+     * @see StorageOperationException
```
Is this needed? I noticed the same issue in several other places too.

FerdiHS commented on src/main/java/anya/parser/Parser.java:
```diff
@@ -0,0 +1,251 @@
+     * @return The command represented by the user input, or {@code Command.UNKNOWN} if the input is not recognized.
+     */
+    public Command parseCommand(String input) {
+        if (input.equals("bye")) {
+            return Command.BYE;
+        }
+        if (input.equals("list")) {
+            return Command.LIST;
+        }
+        if (input.equals("mark")) {
+            return Command.MARK;
+        }
+        if (input.equals("unmark")) {
+            return Command.UNMARK;
+        }
+        if (input.equals("todo")) {
+            return Command.TODO;
+        }
+        if (input.equals("deadline")) {
+            return Command.DEADLINE;
+        }
+        if (input.equals("event")) {
+            return Command.EVENT;
+        }
+        if (input.equals("delete")) {
+            return Command.DELETE;
+        }
+        if (input.equals("find")) {
+            return Command.FIND;
+        }
```
Shouldn't using switch case would be better?

FerdiHS commented on src/main/java/anya/storage/Storage.java:
```diff
@@ -0,0 +1,197 @@
+    /**
+     * Signals that the given file path does not fulfill the storage filepath constraints.
+     */
+    public static class InvalidStorageFilePathException extends Exception {
+        public InvalidStorageFilePathException(String message) {
+            super(message);
+        }
+    }
+
+    /**
+     * Signals that some error has occurred while trying to convert and read/write data between the application
+     * and the storage file.
+     */
+    public static class StorageOperationException extends Exception {
+        public StorageOperationException(String message) {
+            super(message);
+        }
+    }
+
+    /**
+     * Signals that an unknown task type was encountered during conversion.
+     */
+    public static class UnknownTaskException extends Exception {
+        public UnknownTaskException(String message) {
+            super(message);
+        }
+    }
+}
```
Isn't it better to separate Java classes?

FerdiHS commented on src/main/java/anya/task/Task.java:
```diff
@@ -0,0 +1,59 @@
+        this.description = description;
+        this.isDone = false;
+    }
+
```
Shouldn't there be a JavaDoc comment for this method?  I noticed the same issue in several other places too.

FerdiHS commented on text-ui-test/input.txt:
Shouldn't there test cases for other features such as find and delete?

FerdiHS reviewed (commented):
Overall it is very good. Keep up the good works!

tim-pipi committed:
Update javadoc

tim-pipi committed:
Merge branch 'branch-A-CheckStyle'

tim-pipi committed:
Update README

tim-pipi committed:
Set up JavaFX using Gradle

tim-pipi committed:
Set up GUI using JavaFX

tim-pipi committed:
Add Echo functionality

tim-pipi committed:
Add Dialog Boxes

tim-pipi committed:
Add custom behavior to DialogBox

tim-pipi committed:
Add GUI

tim-pipi committed:
Merge branch 'branch-Level-10'

tim-pipi committed:
Add Varargs for add in Task

tim-pipi committed:
Merge branch 'branch-A-Varargs'

LHeng1 commented on README.md:
````diff
@@ -1,24 +1,36 @@
-   ```
+# Anya Forger - Your Task Management Chat bot
+
+Anya Forger is a chat bot inspired by the character from the [_Spy x Family_](https://en.wikipedia.org/wiki/Spy_%C3%97_Family) anime. She is your ultimate task management companion. Whether you're a secret agent or just someone with a busy schedule, Anya can help you stay organized and on top of your tasks.
````
![Amazed_Anya](https://github.com/nus-cs2103-AY2324S1/ip/assets/110957690/08b4c455-591f-48db-ae6e-cd57d2e99e56)
I like how you created your own design instead of Duke! I am a big fan of the anime as well!

LHeng1 commented on src/main/java/anya/Anya.java:
```diff
@@ -0,0 +1,77 @@
+package anya;
+
+import java.io.IOException;
+
+import anya.parser.Parser;
+import anya.storage.Storage;
+import anya.storage.Storage.InvalidStorageFilePathException;
+import anya.storage.Storage.StorageOperationException;
+import anya.task.TaskList;
+import anya.ui.Ui;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.VBox;
```
I like how you listed down imported classes/packages explicitly.

LHeng1 commented on src/main/java/anya/parser/Parser.java:
```diff
@@ -0,0 +1,253 @@
+                        System.out.println("    Waku waku! I've added this task:\n" + t);
+                        System.out.println("    Now you have " + tasks.size() + " tasks in the list!");
+                        break;
+                    }
+                    case DELETE: {
+                        // Error: No argument or Multiple arguments provided
+                        if (details.isEmpty() || details.split(" ").length != 1) {
+                            throw new InvalidArgumentException("☹ Waku waku! Please input in the following format:\n"
+                                    + "    delete <taskNumber>");
+                        }
+                        // Error: Argument provided is not a number
+                        try {
+                            Integer.parseInt(details);
+                        } catch (NumberFormatException e) {
+                            throw new InvalidArgumentException("☹ Waku waku! "
+                                    + "Please only input INTEGERs after the word unmark!");
+                        }
+                        // Error: Argument provided is not within anya.task numbers
+                        int taskNumber = Integer.parseInt(details);
+                        if (taskNumber < 1 || taskNumber > tasks.size()) {
+                            throw new InvalidArgumentException("☹ Waku waku! "
+                                    + "I don't see a anya.task with the number:" + taskNumber);
+                        }
+
+                        Task t = tasks.get(taskNumber - 1);
+                        tasks.remove(t);
+
+                        String result = "    Waku waku! I've removed this task:\n" + t;
+                        result += "    Now you have " + tasks.size() + " tasks in the list!";
+                        System.out.println(result);
+                        break;
+                    }
+                    case FIND: {
+                        tasks.find(details);
+                        break;
+                    }
+                    default:
+                        String result = "☹ Waku waku!!! I'm sorry, but I don't know what that means (yet) :( ";
+                        System.out.println(result);
+                }
```
Should there be indentation for the `case` clauses in a _switch_ statement? 

If you are using IntelliJ, perhaps you can open file settings, go to **Editor > CodeStyle > Java > Wrapping and Braces, at 'switch' statement, untick the box 'Indent case branches'** 👍 

Additional tip: To reformat a section of code in Intellij, can refer to this [guide](https://www.jetbrains.com/idea/guide/tips/reformat-file/#:~:text=Sometimes%20code%20formatting%20can%20get,L%20(Windows%2FLinux))!

LHeng1 commented on src/main/java/anya/task/TaskList.java:
```diff
@@ -0,0 +1,92 @@
+ * The `TaskList` class provides methods for adding, removing, accessing, and listing tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
```
I like how you used plural form for representing collections 👍

LHeng1 reviewed (commented):
Overall I found your code easy to read for the most part except the _switch_ part where it was quite long. Perhaps you could consider splitting it into smaller classes if it makes sense. Other than that LGTM!

tim-pipi committed:
Refactor UI and Messages

tim-pipi committed:
Refactor GUI Code

The getResponse() function is only a basic echo function.

A working GUI should have the same functionality as the text UI.

Let's,
* update the getResponse() function
* ensure code quality in the updated functions

tim-pipi committed:
Merge pull request #1 from tim-pipi/branch-Level-10

Branch level 10

tim-pipi committed:
Update build.gradle to fix JavaFX issue

The main class should be Launcher

Running other classes will give a runtime error

Let's update the main class in build.gradle

tim-pipi committed:
Add assert in Parser.java

tim-pipi committed:
Refactor for Code Quality

tim-pipi committed:
Merge pull request #3 from tim-pipi/branch-A-Assertions

Branch a assertions

tim-pipi committed:
Merge branch 'master' into branch-A-CodeQuality

tim-pipi committed:
Merge pull request #4 from tim-pipi/branch-A-CodeQuality

Refactor for Code Quality

tim-pipi committed:
Add Help functionality

tim-pipi committed:
Merge pull request #5 from tim-pipi/branch-C-Help

Add Help functionality

tim-pipi committed:
Update User Guide

tim-pipi committed:
Merge branch 'branch-A-UserGuide'

tim-pipi committed:
Update README
